Realm/ObjectStore/object_store.cpp

```diff
@@ -486,6 +515,65 @@ void ObjectStore::delete_data_for_object(Group *group, StringData object_type) {
     }
 }
 
+void ObjectStore::rename_property(Group *group, Schema& passed_schema, StringData object_type, StringData old_name, StringData new_name) {
+    Schema schema = schema_from_group(group);
+    auto matching_schema = schema.find(object_type);
+    if (matching_schema == schema.end()) {
+        throw PropertyRenameMissingOldObjectTypeException(object_type);
+    }
+    Property *old_property = matching_schema->property_for_name(old_name);
+    if (old_property == nullptr) {
+        throw PropertyRenameMissingOldPropertyException(old_name, new_name);
+    }
+    TableRef table = table_for_object_type(group, object_type);
+    if (!table) {
+        throw PropertyRenameMissingNewObjectTypeException(object_type);
```

Comment:
This line isn't reachable (it's effectively the same check as the `matching_schema == schema.end()` one above, but it throws a different exception...).